src/controller/JCMCommissioner.cpp

```diff
@@ -0,0 +1,438 @@
+        using namespace Clusters::JointFabricAdministrator::Attributes;
+        AdministratorFabricIndex::TypeInfo::DecodableType administratorFabricIndex;
+
+        VerifyOrReturnError(path.mAttributeId == AdministratorFabricIndex::Id, CHIP_NO_ERROR);
+        ReturnErrorOnFailure(attributeCache->Get<AdministratorFabricIndex::TypeInfo>(path, administratorFabricIndex));
+
+        if (!administratorFabricIndex.IsNull() && administratorFabricIndex.Value() != kUndefinedFabricIndex)
+        {
+            ChipLogProgress(Controller, "JCM: AdministratorFabricIndex: %d", administratorFabricIndex.Value());
+            mInfo.adminFabricIndex = administratorFabricIndex.Value();
+            mInfo.adminEndpointId = path.mEndpointId;
+        }
+        else
+        {
+            ChipLogError(Controller, "JCM: JF Administrator Cluster not found!");
+            return CHIP_ERROR_NOT_FOUND;
+        }
+
+        return CHIP_NO_ERROR;
+    });
+
+    return err;
+}
+
+CHIP_ERROR JCMDeviceCommissioner::ParseOperationalCredentials(ReadCommissioningInfo & info)
+{
+    auto attributeCache = info.attributes;
+
+    CHIP_ERROR err = attributeCache->ForEachAttribute(OperationalCredentials::Id, [this, &attributeCache](const ConcreteAttributePath & path) {
+        using namespace chip::app::Clusters::OperationalCredentials::Attributes;
+
+        switch (path.mAttributeId)
+        {
+            case Fabrics::Id: {
+                Fabrics::TypeInfo::DecodableType fabrics;
+                ReturnErrorOnFailure(attributeCache->Get<Fabrics::TypeInfo>(path, fabrics));
+
+                auto iter = fabrics.begin();
+                while (iter.Next())
+                {
+                    auto & fabricDescriptor = iter.GetValue();
+                    if (fabricDescriptor.fabricIndex != kUndefinedFabricIndex)
+                    {
+                        if (fabricDescriptor.rootPublicKey.size() != Crypto::kP256_PublicKey_Length)
+                        {
+                            ChipLogError(Controller, "JCM: DeviceCommissioner::ParseJFAdministratorInfo - fabric root key size mismatch");
+                            return CHIP_ERROR_KEY_NOT_FOUND;
+                        }
+
+                        mInfo.rootKeySpan = fabricDescriptor.rootPublicKey;
+                        mInfo.adminVendorId = fabricDescriptor.vendorID;
+                        mInfo.adminFabricId = fabricDescriptor.fabricID;
+
+                        if (fabricDescriptor.VIDVerificationStatement.HasValue())
+                        {
+                            ChipLogError(Controller, "JCM: Per-home RCAC are not supported by JF for now!");
+                            return CHIP_ERROR_CANCELLED;
+                        }
+                        ChipLogProgress(Controller, "JCM: Successfully parsed the Administrator Fabric Table");
+                        break;
+                    }
+                }
+                return CHIP_NO_ERROR;
```

Comment:
So this returns CHIP_NO_ERROR if the Fabrics attribute was present in the cache but malformed, right?  Is that the desired behavior?

Reply:
Applied Doru's fix for this issue to this branch.